src/app/templates/TopUpInput/StaticCurrencyImage.tsx

```diff
@@ -12,10 +12,21 @@ interface Props {
   imageSrc?: string;
   fitImg?: boolean;
   className?: string;
+  isVisible?: boolean;
 }
 
-export const StaticCurrencyImage: FC<Props> = ({ currencyCode, isFiat, imageSrc, fitImg, className }) => {
+export const StaticCurrencyImage: FC<Props> = ({
+  currencyCode,
+  isFiat,
+  imageSrc,
+  fitImg,
+  className,
+  isVisible = true
+}) => {
   const [isFailed, setIsFailed] = useState(false);
+  const [wasVisible, setWasVisible] = useState(isVisible);
+
+  useEffect(() => void (isVisible && !wasVisible && setWasVisible(true)), [isVisible, wasVisible]);
```

Comment:
This code is not correct because it sets `wasVisible` to `true` after the first render although `isVisible` is `false`. However, `wasVisible` dependency can be easily removed